provider/k8s/metric_scraper.go

```diff
@@ -0,0 +1,144 @@
+package k8s
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/convox/convox/pkg/structs"
+	"github.com/pkg/errors"
+	am "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type MetricScraperClient struct {
+	c       *http.Client
+	cluster kubernetes.Interface
+	host    string
+}
+
+func NewMetricScraperClient(cluster kubernetes.Interface, host string) *MetricScraperClient {
+	return &MetricScraperClient{
+		c: &http.Client{
+			Timeout: 10 * time.Second,
+		},
+		cluster: cluster,
+		host:    host,
+	}
+}
+
+func (m *MetricScraperClient) GetRackMetrics(opts structs.MetricsOptions) (structs.Metrics, error) {
+	if m.host == "" {
+		return nil, errors.WithStack(fmt.Errorf("unimplemented"))
+	}
+
+	ns, err := m.cluster.CoreV1().Nodes().List(am.ListOptions{})
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	nodeNames := []string{}
+	var cpuAllocatable, memAllocatable float64
+	for _, n := range ns.Items {
+		nodeNames = append(nodeNames, n.ObjectMeta.Name)
+		cpuAllocatable += float64(n.Status.Allocatable.Cpu().MilliValue())
+		memAllocatable += float64(n.Status.Allocatable.Memory().Value())
+	}
+
+	cpus, err := m.GetNodesMetrics(strings.Join(nodeNames, ","), structs.ScraperMetricTypeCpu)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	if len(cpus.Items) == 0 {
+		return nil, fmt.Errorf("metrics unavailable")
+	}
+
+	for i := 1; i < len(cpus.Items); i++ {
+		for j, d := range cpus.Items[i].MetricPoints {
+			if j < len(cpus.Items[0].MetricPoints) {
+				cpus.Items[0].MetricPoints[j].Value += d.Value
+			}
+		}
+	}
+
+	cpum := structs.Metric{
+		Name: "cluster:cpu:utilization",
+	}
+	for _, d := range cpus.Items[0].MetricPoints {
+		p := caculatePercentage(float64(d.Value), cpuAllocatable)
+		cpum.Values = append(cpum.Values, structs.MetricValue{
+			Average: p,
```

Comment:
in console3 ui, we show the usage in %. For v2 rack, it already receives the data in percentage format, so for that consistency I also send in percentage format.

`Count: 1` here means number of metrics points it has aggregated, since it is first one so value is 1. You can see that metrics points aggregated by time period. https://github.com/convox/convox/blob/feat-460_v3_rack_metrics/provider/k8s/helpers.go#L353